api/src/pcapi/core/mails/transactional/users/ubble/reminder_emails.py

```diff
@@ -53,10 +53,13 @@ def send_reminder_emails() -> None:
             logger.error("Could not find reminder email template for reason code %s", relevant_reason_code)
             continue
 
-        users_to_notify.append(user.id)
+        if relevant_reason_code not in users_to_notify_per_code:
+            users_to_notify_per_code[relevant_reason_code] = [user.id]
+        else:
+            users_to_notify_per_code[relevant_reason_code].append(user.id)
```

Comment:
(style) Tu peux définir `users_to_notify` comme un `collections.defaultdict(list)`. Cela te permet d'écrire ensuite simplement : 

```suggestion
        users_to_notify_per_code[relevant_reason_code].append(user.id)
```